src/main/java/de/picturesafe/search/elasticsearch/impl/mapping/MappingResolver.java

```diff
@@ -0,0 +1,166 @@
+        final boolean sortableAndAggregatable = isSortableAndAggregatable(properties);
+        return StandardFieldConfiguration.builder(name, type)
+                    .sortable(sortableAndAggregatable)
+                    .aggregatable(sortableAndAggregatable)
+                    .analyzer(stringValue(properties, "analyzer"))
+                    .copyTo((Collection<String>) properties.get("copy_to"))
+                    .additionalParameters(additionalParameters(properties))
+                    .withoutIndexing(!enabled)
+                    .build();
+    }
+
+    private static MappingField resolveObjectField(String name, Map<String, Object> doc) {
+        final List<Locale> locales = new ArrayList<>();
+        final MutableObject<FieldConfiguration> mutableFieldConfig = new MutableObject<>();
+        objectValue(doc, "properties").forEach((subName, properties) -> {
+            final Locale locale = locale(subName);
+            if (locale != null) {
+                locales.add(locale);
+            }
+            final MappingField field = resolveField(subName, (Map<String, Object>) properties);
+            if (mutableFieldConfig.getValue() == null) {
+                mutableFieldConfig.setValue(field.getFieldConfiguration());
+            } else if (!field.getFieldConfiguration().equalsBesidesName(mutableFieldConfig.getValue())) {
+                throw new RuntimeException("Object fields are only supported as multilingual text fields or with enabled=false!");
+            }
+        });
+        final FieldConfiguration fieldConfiguration = StandardFieldConfiguration.builder(name, mutableFieldConfig.getValue()).multilingual(true).build();
+        return new MappingField(fieldConfiguration, locales);
+    }
+
+    private static Locale locale(String name) {
+        try {
+            return LocaleUtils.toLocale(name);
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    private static MappingField resolveNestedField(String name, Map<String, Object> doc) {
+        final List<StandardFieldConfiguration> nestedFields = new ArrayList<>();
+        objectValue(doc, "properties").forEach((subName, properties) -> {
+            final MappingField field = resolveField(subName, (Map<String, Object>) properties);
+            final FieldConfiguration fieldConfiguration = field.getFieldConfiguration();
+            if (fieldConfiguration instanceof StandardFieldConfiguration) {
+                nestedFields.add((StandardFieldConfiguration) fieldConfiguration);
+            } else {
+                throw new RuntimeException(("Unsupported inner field configuration for nested field: " + fieldConfiguration));
+            }
+        });
+        final FieldConfiguration fieldConfiguration
+                = StandardFieldConfiguration.builder("article", ElasticsearchType.NESTED).nestedFields(nestedFields).build();
```

Comment:
Please replace **"article"** with argument 'name'

Reply:
Oops, the devilish c&p again!